pkg/upload/js.go

```diff
@@ -16,6 +24,158 @@ type JsOptions struct {
 	Path        utils.Paths `arg:"" name:"path" help:"Path to a directory of source maps and bundles to upload" type:"path" default:"."`
 }
 
+// Resolve the project root by walking up from the path specified until a package.json is found
+func resolveProjectRoot(jsOptions JsOptions, path string) (string, error) {
```

Comment:
I think we might want to separate out concerns here a bit if the user has explicitly provided a project root, then we should use this string as the projectRoot parameter to the API exactly. I don't think we should conflate this concept with attempts to find the `package.json`; nor resolve the path or navigate for the purposes of the project root value.

In order to find a potential `package.json` I suggest we only look in the current directory, rather than risk finding config files at a higher level (e.g. in a monorepo). I think the principle of least surprise is better here.

Reply:
If the user has specifically provided a project root then that is used exactly (see the lines below this). The behaviour of walking up to find the package.json is only used as a fallback.

The bugsnag-source-map CLI uses the working directory as the project root if one is not otherwise specified. We could copy that behaviour if you prefer?

Reply:
Let's stick to the current working directory for both project root default and potential location for the `package.json`.

Reply:
The bugsnag-source-maps CLI recursively searches upwards to find a package.json in order to find the version. Is this not the desired behaviour here?

Reply:
OK - if it's existing behaviour on the old tool, I'm happy to keep it for the package.json search 👍